### ISSUE TITLE
Critical error on startup / when adding an alarm

### ISSUE THREAD

**Issue by NathanBnm** [sgpthomas/hourglass#71]
Here is the error I get: `GDBus.Error:org.freedesktop.DBus.Error.ServiceUnknown: The name com.github.sgpthomas.hourglass was not provided by any .service files (g-dbus-error-quark, 2)`

Full log here:

```
CRITICAL **: 12:22:49.471: hourglass_daemon_saved_alarms_get_alarms: assertion 'self != NULL' failed
[FATAL 12:22:54.891339] file /home/nathan/Téléchargements/hourglass/src/Widgets/AlarmTimeWidget.vala: line 196: uncaught error: GDBus.Error:org.freedesktop.DBus.Error.ServiceUnknown: The name com.github.sgpthomas.hourglass was not provided by any .service files (g-dbus-error-quark, 2)
```

**Comment by NathanBnm** [comment]
Related to #72 

When handling errors the app is not starting (which is normal because the app cannot work if the daemon is not started correctly).

When handling errors (see #72) here is the log I get, which is more easy to read:

```
$ com.github.sgpthomas.hourglass
[DEBUG 12:24:51.944705] Settings.vala:164: Loading settings from schema 'com.github.sgpthomas.hourglass.saved'
[DEBUG 12:24:51.945329] Settings.vala:164: Loading settings from schema 'org.gnome.desktop.interface'
[INFO 12:24:51.945505] Hourglass.vala:81: starting daemon
[ERROR 12:24:51.951954] DBusManager.vala:51: GDBus.Error:org.freedesktop.DBus.Error.ServiceUnknown: The name com.github.sgpthomas.hourglass was not provided by any .service files
Trappe pour point d'arrêt et de trace
Message: 12:24:51.966: HourglassDaemon.vala:39: Hourglass-Daemon started
** Message: 12:24:51.967: AlarmManager.vala:28: Initiating Alarm Manager

** (process:5985): CRITICAL **: 12:24:51.967: hourglass_daemon_saved_alarms_get_alarms: assertion 'self != NULL' failed
```

**Comment by ryonakano** [comment]
I can't reproduce this issue―neither on the latest master or your `fix-warnings` branch.

<details>
<summary>FYI my log</summary>

```
ryo@ryo-pc:~/Projects/hourglass$ com.github.sgpthomas.hourglass
[DEBUG 19:25:32.462392] [GLib-GIO] _g_io_module_get_default: Found default implementation dconf (DConfSettingsBackend) for ‘gsettings-backend’
[DEBUG 19:25:32.462546] Settings.vala:164: Loading settings from schema 'com.github.sgpthomas.hourglass.saved'
[DEBUG 19:25:32.462801] Settings.vala:164: Loading settings from schema 'org.gnome.desktop.interface'
[INFO 19:25:32.462881] Hourglass.vala:81: starting daemon
[DEBUG 19:25:32.462912] [GLib] posix_spawn avoided (automatic reaping requested) (workdir specified) (fd close requested) 
[DEBUG 19:25:32.466849] [GLib] unsetenv() is not thread-safe and should not be used after threads are created
[INFO 19:25:32.468781] StyleManager.vala:30: Loaded /com/github/sgpthomas/hourglass/style/text.css
[INFO 19:25:32.469600] StyleManager.vala:30: Loaded /com/github/sgpthomas/hourglass/style/elements.css
** Message: 19:25:32.477: HourglassDaemon.vala:39: Hourglass-Daemon started
** Message: 19:25:32.477: AlarmManager.vala:28: Initiating Alarm Manager

** (process:62973): CRITICAL **: 19:25:32.477: hourglass_daemon_saved_alarms_get_alarms: assertion 'self != NULL' failed
** Message: 19:25:32.484: HourglassDaemon.vala:63: Daemon Activated

** (com.github.sgpthomas.hourglass-daemon:62973): CRITICAL **: 19:25:32.484: HourglassServer.vala:78: Could not aquire service name
```

</details>

**Comment by ryonakano** [comment]
Ah OK, I can reproduce by *killalling* the daemon

**Comment by NathanBnm** [comment]
Seems that for me the daemon is not starting

**Comment by ryonakano** [comment]
On my side, only the first launch I face this error and the UI is not shown. After that, the daemon starts correctly and the UI is shown.

**Comment by NathanBnm** [comment]
> On my side, only the first launch I face this error and the UI is not shown. After that, the daemon starts correctly and the UI is shown.

This does not work on my side :confused:

**Comment by ryonakano** [comment]
Could you check if #77 fixes your issue?

**Comment by NathanBnm** [comment]
@ryonakano unfortunately not. I still have the following issue:
```
DBusManager.vala:51: GDBus.Error:org.freedesktop.DBus.Error.ServiceUnknown: The name com.github.sgpthomas.hourglass was not provided by any .service files
```

**Comment by ryonakano** [comment]
OK, I tested `master` on the latest fresh Hera and Odin daily, and found this issue seems to be reproducible only on Hera. So, I guess these errors are due to DBus on Hera/Bionic.

**Comment by NathanBnm** [comment]
Humm I'm running the latest version elementary OS Hera. What is Odin?

**Comment by ryonakano** [comment]
That's the codename of the next version which is based on Ubuntu 20.04.

**Comment by NathanBnm** [comment]
Okay didn't know it was already available

**Comment by Gameransari** [comment]
> Here is the error I get: `GDBus.Error:org.freedesktop.DBus.Error.ServiceUnknown: The name com.github.sgpthomas.hourglass was not provided by any .service files (g-dbus-error-quark, 2)`
> 
> 
> 
> Full log here:
> 
> 
> 
> ```
> 
> CRITICAL **: 12:22:49.471: hourglass_daemon_saved_alarms_get_alarms: assertion 'self != NULL' failed
> 
> [FATAL 12:22:54.891339] file /home/nathan/Téléchargements/hourglass/src/Widgets/AlarmTimeWidget.vala: line 196: uncaught error: GDBus.Error:org.freedesktop.DBus.Error.ServiceUnknown: The name com.github.sgpthomas.hourglass was not provided by any .service files (g-dbus-error-quark, 2)
> 
> ```

The Solution is here https://youtu.be/34iXHWuhsss

**Comment by Gameransari** [comment]
**Everyone's should try this **
This might be helpful https://youtu.be/34iXHWuhsss

**Comment by ryonakano** [comment]
elementary OS does not come with `xfce4-panel`, so that does not work. The error message is the same, but the causes would be different.

**Comment by ryonakano** [comment]
Should be fixed with the latest release 1.4.0 for Odin.
